File: app/src/main/java/com/aaron/yespdf/about/IAboutContract.java
package com.aaron.yespdf.about;

import java.util.List;

/**
 * @author Aaron aaronzheng9603@gmail.com
 */
interface IAboutContract {

    interface V<A, B> {
        void attachPresenter();

        void onShowMessage(List<A> list);

        void onShowLibrary(List<B> list);
    }

    interface P {
        void detachView();

        void requestMessage(int[] iconId, String[] title);

        void requestLibrary(String[] name, String[] author, String[] introduce);
    }
}

File: app/src/main/java/com/aaron/yespdf/about/IAboutContract.kt
package com.aaron.yespdf.about

import com.aaron.yespdf.common.IPresenter
import com.aaron.yespdf.common.IView

/**
 * @author Aaron aaronzheng9603@gmail.com
 */
interface IAboutView : IView {
    fun onShowMessage(list: List<Message>)
    fun onShowLibrary(list: List<Library>)
}

abstract class IAboutPresenter(view: IAboutView) : IPresenter<IAboutView>(view) {
    abstract fun requestMessage(iconId: IntArray, title: Array<String>)
    abstract fun requestLibrary(name: Array<String>, author: Array<String>, introduce: Array<String>)
}